controller/instance_manager_controller.go

```diff
@@ -585,14 +585,21 @@ func (imc *InstanceManagerController) canDeleteInstanceManagerPDB(im *longhorn.I
 		}
 
 		hasHealthyReplicaOnAnotherNode := false
+		isUnusedReplicaOnCurrentNode := false
 		for _, r := range replicas {
 			if r.Spec.HealthyAt != "" && r.Spec.FailedAt == "" && r.Spec.NodeID != imc.controllerID {
 				hasHealthyReplicaOnAnotherNode = true
 				break
 			}
+			// Fix drained node `pod's disruption budget` issue, when LHV is is created through Longhorn UI,
+			// but has not yet attached and replicated.
+			if r.Spec.HealthyAt == "" && r.Spec.FailedAt == "" && r.Spec.NodeID == imc.controllerID {
```

Comment:
```
If a replica has never been started, there is no data stored in this replica, and retaining it makes no sense for HA.
Hence Longhorn doesn't need to block the PDB removal for the replica.
This case typically happens on a newly created volume that hasn't been attached to any node.
```

You can add the Github link if you prefer.

Reply:
changed